htlcswitch/failure.go

```diff
@@ -9,6 +9,16 @@ import (
 	"github.com/lightningnetwork/lnd/lnwire"
 )
 
+// RouteHtlcError is an interface which is implemented by errors obtained when
+// trying to route a HTLC. A routing error can occur at our node, when we fail
+// a HTLC on the incoming/outgoing link or further down the line.
+type RouteHtlcError interface {
```

Comment:
Could also be called `KnownError` or `ClearTextError` since this covers all cases where we have a non-opaque error?